lib/Doctrine/ODM/MongoDB/Iterator/BaseIterator.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Iterator;
+
+use Generator;
+use Traversable;
+
+/**
+ * Iterator for wrapping a Traversable and caching its results.
+ *
+ * By caching results, this iterators allows a Traversable to be counted and
+ * rewound multiple times, even if the wrapped object does not natively support
+ * those operations (e.g. MongoDB\Driver\Cursor).
+ *
+ * @internal
+ */
+final class BaseIterator implements Iterator
+{
+    private $key;
+    
+    private $current;
+
+    /** @var Generator */
```

Comment:
```suggestion
    /** @var Generator|null */
```

This will fix the current PHPStan errors in this file, but most likely cause others. You can use the `getIterator()` method from `CachingIterator` to prevent these errors.